src/Azure.DataApiBuilder.Mcp/BuiltInTools/CreateRecordTool.cs

```diff
@@ -172,31 +171,49 @@ public async Task<CallToolResult> ExecuteAsync(
                 {
                     return new CallToolResult
                     {
-                        Content = [new TextContentBlock 
-                        { 
-                            Type = "text", 
+                        Content = [new TextContentBlock
+                        {
+                            Type = "text",
                             Text = $"Successfully created record in entity '{entityName}'. Result: {JsonSerializer.Serialize(createdResult.Value)}"
                         }]
                     };
                 }
                 else if (result is ObjectResult objectResult)
                 {
-                    // Check if this is an error status code (400+ range)
+                    bool isError = objectResult.StatusCode.HasValue && objectResult.StatusCode.Value >= 400 && objectResult.StatusCode.Value != 403;
+                    string text = isError
+                        ? $"Failed to create record in entity '{entityName}'. Error: {JsonSerializer.Serialize(objectResult.Value)}"
+                        : $"Successfully created record in entity '{entityName}'. Result: {JsonSerializer.Serialize(objectResult.Value)}. Unable to perform " +
+                        $"read-back of inserted records";
                     return new CallToolResult
```

Comment:
Should use BuildErrorResult in case of `isError` otherwise use BuildSuccessResult